src/components/Board.js

```diff
@@ -0,0 +1,214 @@
+
+import './Board.css';
+import Card from './Card';
+import NewCardForm from './NewCardForm';
+
+//const BOARD_STATES = ['DISPLAY', 'ADD', 'DELETE'];
+
+
+class Board extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      mode: 'DISPLAY',
+      cards: [],
+      message: '',
+    };
+  }
+
+  onAddClick = (event) => {
+    event.preventDefault();
+    this.setState({
+      mode: 'ADD',
+    });
+  }
+
+  componentDidMount() {
+    this.setState({
+      message: 'Loading Cards',
+    });
+
+    axios.get(`${this.props.url}${this.props.boardId}/cards`)
+    .then((response) => {
+      const cards = [];
+      response.data.forEach((card) => {
+        cards.push({
+          text: card.card.text,
+          image_url: card.card.image_url,
+          id: card.card.id,
+        });
+      });
+      this.setState({
+        cards,
+        message: '',
+      });
+    })
+    .catch((response) => {
+      this.setState({
+        message: `Error: ${response.data}`,
+      })
+    });
+  }
+
+  getCards = () => {
+    return this.state.cards.map((card, index) => {
+      return (
+        <Card
+          key={index}
+          text={card.text}
+          image_url={card.image_url}
+          deleteCardCallback={this.deleteCard}
+          id={card.id === undefined ? -1: card.id}
+          />
+      );
+    });
+  }
+
+
+
+  renderFieldValidationErrors = (field, errorArray) => {
+    const list = errorArray.map((message, index) => {
+      return (
+        <li key={`${field}-${index}`}>
+          {message}
+        </li>
+      )
+    });
+    return (
+      <div>
+        <strong>{field}:</strong>
+        <ul className="validation-errors">
+          {list}
+        </ul>
+      </div>
+    );
+  }
+
+  renderValidationErrors = (responseData) => {
+    if (responseData['errors']) {
+
+      const validationErrors = responseData['errors']
+      const errorList = Object.keys(validationErrors).map((field) => {
+        return (
+          <div key={`${field}`}>
+            {this.renderFieldValidationErrors(field, validationErrors[field])}
+          </div>
+        );
+      });
+      return (
+        <div>
+          {errorList}
+        </div>
+      );
+    }
+    else {
+      return (
+        <div />
+      )
+    }
+  }
+
+  deleteCard = (cardId) => {
+    const cards = this.state.cards.filter((card) => card.id !== parseInt(cardId, 10))
+    this.setState({
+      cards,
+    });
+    axios.delete(`${this.props.url}${this.props.boardId}/cards/${cardId}`)
+    .then(() => {
+      this.setState({
+        message: 'Card deleted',
+      })
+    })
+    .catch(() => {
+      this.setState({
+        message: 'Cannot delete card',
+      });
+    });
+  }
+
+
+  addCard = (card) => {
+
+    axios.post(`${this.props.url}${this.props.boardId}/cards`, card)
+    .then((response) => {
+      cards.forEach((card) => {
+        if (card.id === undefined && card.text === response.data.card.text) {
```

Comment:
This `forEach` construct to search for the correct card to update, based on the card text, seems unnecessary.